PurchasesTests/Networking/BackendTests.swift

```diff
@@ -1082,35 +1014,32 @@ class BackendTests: XCTestCase {
         backend?.postReceiptData(receiptData,
                                  appUserID: userID,
                                  isRestore: isRestore,
-                                 productIdentifier: nil,
-                                 price: nil,
-                                 paymentMode: RCPaymentMode.none,
-                                 introductoryPrice: nil,
-                                 currencyCode: nil,
-                                 subscriptionGroup: nil,
-                                 discounts: nil,
+                                 productInfo: nil,
                                  presentedOfferingIdentifier: nil, observerMode:
                                  observerMode, subscriberAttributes:
                                  nil, completion:
                                  { (purchaserInfo, error) in
             completionCalled += 1
         })
 
-        let discount = RCPromotionalOffer.init()
+        let discount = RCPromotionalOffer()
         discount.offerIdentifier = "offerid"
-        discount.paymentMode = RCPaymentMode.payAsYouGo
+        discount.paymentMode = .payAsYouGo
         discount.price = 12
-
+        let productInfo = RCProductInfo(productIdentifier: "product_id",
+                                        paymentMode: .none,
+                                        currencyCode: "UYU",
+                                        price: 15.99,
```

Comment:
yeah, I did think about that when writing. the tests are definitely more noisy this way. 
The reason I went this way was: 
- I wanted _only_ arguments that were truly nullable (i.e.: you might not be able to extract them from an SKProduct) to be nullable. 
- I didn't want properties to be editable unless they needed to (this part could be done by doing a protected header, though)
- Even though you can initialize RCProductInfo with no arguments (because it inherits from NSObject), that can lead to some pretty bad gotchas with swift: 

When you initialize with no arguments, everything is nil. However, swift will assume that the object can't be nil unless nullable is specified (or no nullability is specified in the entire file). 

For example, if I do: 
```swift
let productInfo = RCProductInfo()
let price = productInfo.price
```
That will crash, because price is nil, but swift didn't expect that. I won't be able to even check for it, since `if price != nil` will raise a compiler warning saying that price can't be nil. 

To get around this, I usually disable the no-params constructor, I forgot about it this time. 
To disable it, you do `NS_UNAVAILABLE` ([example here](https://github.com/RevenueCat/purchases-ios/blob/develop/Purchases/Misc/RCSystemInfo.h#L16) which will raise a compiler warning when using that constructor. 

That way you have control over the properties to make sure that they're never called from swift without having initialized them properly. 

Thinking about possible fixes for this, I think the best workaround is to have a protected header that allows for these properties to be modified, and then a single function that always creates a valid ProductInfo object, so that I can do: 
```swift
let productInfo = createProductInfo()
productInfo.whateverProperty = "the value I want to test"
```

I'll do that refactor

Reply:
done!